2nd-gen/alerts-to-discord/functions/index.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// [START v2import]
+const {
+  onNewFatalIssuePublished,
+} = require("firebase-functions/v2/alerts/crashlytics");
+const logger = require("firebase-functions/logger");
+// [END v2import]
+
+/**
+ * Posts a message to Discord with Discord's Webhook API
+ *
+ * @param {string} botName - The bot username to display
+ * @param {string} messageBody - The message to post (Discord MarkDown)
+ */
+async function postMessageToDiscord(botName, messageBody) {
+  const webhookUrl = process.env.DISCORD_WEBHOOK_URL;
+  if (!webhookUrl) {
+    throw new Error(
+        "No webhook URL found. Set the Discord Webhook URL before deploying. Learn more about Discord webhooks here: https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks",
+    );
+  }
+
+  // node-fetch needs a special import: https://github.com/node-fetch/node-fetch#commonjs
+  const fetch = await import("node-fetch");
+
+  return fetch(webhookUrl, {
+    method: "POST",
+    headers: {"Content-Type": "application/json"},
+    body: JSON.stringify(
+        // Here's what the Discord API supports in the payload:
+        // https://discord.com/developers/docs/resources/webhook#execute-webhook-jsonform-params
+        {
+          content: messageBody,
+          username: botName,
+        },
+    ),
+  });
+}
+
+// [START v2Alerts]
+/**
+ * function triggered by Crashlytics that publishes a message
+ * to Discord whenever a new fatal issue occurs.
+ */
+// [START v2CrashlyticsAlertTrigger]
+exports.postmessagetodiscord = onNewFatalIssuePublished({}, async (event) => {
```

Comment:
If we're not setting an `appId` or any options, then we should just be able to omit the `options` object and use the handler only API, like
```
exports.postmessagetodiscord = onNewFatalIssuePublished(async (event) => {
```